articles/specs/launch.0.1.0.xsd

```diff
@@ -0,0 +1,562 @@
+        defined in place or in a file to be loaded.
+      </xs:documentation>
+    </xs:annotation>
+
+    <xs:complexType>
+      <xs:choice minOccurs="0" maxOccurs="unbounded">
+        <xs:annotation>
+          <xs:documentation xml:lang="en">
+            A collection of nested ROS parameters to be set.
+          </xs:documentation>
+        </xs:annotation>
+
+        <xs:element ref="param"/>
+      </xs:choice>
+      <xs:attribute name="name" type="xs:string" use="required">
+        <xs:annotation>
+          <xs:documentation xml:lang="en">
+            Name of the ROS parameter to be set.
+          </xs:documentation>
+        </xs:annotation>
+      </xs:attribute>
+      <xs:attribute name="value" type="xs:string" use="required">
+        <xs:annotation>
+          <xs:documentation xml:lang="en">
+            Value or list of values to set the the ROS parameter with.
+          </xs:documentation>
+        </xs:annotation>
+      </xs:attribute>
+      <xs:attribute name="sep" type="xs:string" use="optional">
+        <xs:annotation>
+          <xs:documentation xml:lang="en">
+            Value separator when dealing with list of values.
+          </xs:documentation>
+        </xs:annotation>
+      </xs:attribute>
+      <xs:attribute name="from" type="xs:string" use="optional">
+        <xs:annotation>
+          <xs:documentation xml:lang="en">
+            Path to the parameters file to be loaded.
+          </xs:documentation>
+        </xs:annotation>
+      </xs:attribute>
+    </xs:complexType>
+  </xs:element>
+
+  <xs:element name="remap">
+    <xs:annotation>
+      <xs:documentation xml:lang="en">
+        Remaps ROS names for a node.
+      </xs:documentation>
+    </xs:annotation>
+
+    <xs:complexType>
+      <xs:attribute name="from" type="xs:string" use="required">
+        <xs:annotation>
+          <xs:documentation xml:lang="en">
+            Name matching expression to look for replacement candidates.
+          </xs:documentation>
+        </xs:annotation>
+      </xs:attribute>
+      <xs:attribute name="to" type="xs:string" use="required">
+        <xs:annotation>
+          <xs:documentation xml:lang="en">
+            Name replacement expression to replace candidates found.
+          </xs:documentation>
+        </xs:annotation>
+      </xs:attribute>
+    </xs:complexType>
+  </xs:element>
+
+  <xs:element name="node">
+    <xs:annotation>
+      <xs:documentation xml:lang="en">
+        Executes a ROS node executable in a local OS process.
+      </xs:documentation>
+    </xs:annotation>
+
+    <xs:complexType>
+      <xs:choice minOccurs="0" maxOccurs="unbounded">
+        <xs:annotation>
+          <xs:documentation xml:lang="en">
+            A collection of ROS parameters, ROS remappings and/or
+            environment variable settings for the launched ROS node.
+          </xs:documentation>
+        </xs:annotation>
+
+        <xs:element ref="env"/>
+        <xs:element ref="param"/>
```

Comment:
When the node has some parameters, it should also have a node name (if not, an error is raised). I don't know if it's possible to add an assert condition to the schema, which checks if node has the 'name' attribute when it contains a 'param' child.